Xamarin.Forms.Platform.iOS/Renderers/ShellTableViewController.cs

```diff
@@ -40,10 +41,20 @@ void OnStructureChanged(object sender, EventArgs e)
 		public void LayoutParallax()
 		{
 			var parent = TableView.Superview;
-
-			TableView.Frame = parent.Bounds;
+			TableView.Frame = parent.Bounds.Inset(0, SafeAreaOffset);
 			if (_headerView != null)
-				_headerView.Frame = new CGRect(0, _headerOffset, parent.Frame.Width, _headerSize);
+			{
+				_headerView.Frame = new CGRect(0, _headerOffset + SafeAreaOffset, parent.Frame.Width, _headerSize);
+
+				if (_headerOffset < 0)
+				{
+					CAShapeLayer shapeLayer = new CAShapeLayer();
+					CGRect rect = new CGRect(0, _headerOffset * -1, parent.Frame.Width, _headerSize + _headerOffset);
```

Comment:
`_headerSize + _headerOffset` cannot be negative?

Reply:
I don't think so... but adding a check just in case is probably worth while

Reply:
I checked, it may be negative if `FlyoutHeaderBehavior` is `CollapseOnScroll`, but no exception is thrown.